src/pattern-library/components/patterns/DialogPatterns.js

```diff
@@ -59,18 +50,67 @@ const showDialog = ({ container, setDialogIsOpen }) => {
           type="text"
         />
       </div>
-    </Dialog>,
+    </DialogComponent>,
     container
   );
 };
 
 export default function DialogPatterns() {
+  // Extra buttons to use in Dialog, Modal examples
+  const buttons = [
+    <LabeledButton key="maybe" onClick={() => alert('You chose maybe')}>
+      Maybe
+    </LabeledButton>,
+    <LabeledButton
+      key="yep"
+      onClick={() => alert('You chose yep')}
+      variant="primary"
+    >
+      Do it!
+    </LabeledButton>,
+  ];
+
+  // Dialog/Modal state for each of the examples
   const [dialogIsOpen, setDialogIsOpen] = useState(false);
+  const [modalIsOpen, setModalIsOpen] = useState(false);
+  const [confirmModalIsOpen, setConfirmModalIsOpen] = useState(false);
+
   const openDialog = () => {
     setDialogIsOpen(true);
     showDialog({
+      DialogComponent: Dialog,
       container: document.getElementById('#dialog1'),
-      setDialogIsOpen,
+      setIsOpen: setDialogIsOpen,
+      props: {
+        buttons,
+      },
+    });
+  };
+
+  const openModal = () => {
```

Comment:
I recognize that the amount of supporting code for these examples/demos is on the excessive side. I'm hoping that we can simplify this as we go, possibly integrating some of the utility functionality in the client's `prompts` module or something.